packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/BillingManager.java

```diff
@@ -143,9 +143,17 @@ public void run() {
           return;
         }
 
-        BillingFlowParams purchaseParams = BillingFlowParams.newBuilder()
-          .setSkuDetails(skuDetails).setOldSku(oldSku).build();
-        mBillingClient.launchBillingFlow(mActivity, purchaseParams);
+        BillingFlowParams.Builder purchaseParams = BillingFlowParams.newBuilder()
+          .setSkuDetails(skuDetails);
+
+        if (purchaseToken != null) {
```

Comment:
Could we use [`getBuyIntentToReplaceSkus`](https://android-doc.github.io/google/play/billing/billing_reference.html#upgrade-getBuyIntentToReplaceSkus) instead? That would avoid a breaking API change. (We could raise a warning for some versions and silently convert passed SKUs to an array).